geoips/interfaces/base.py

```diff
@@ -0,0 +1,360 @@
+        )
+
+    def get_plugin(self, name):
+        """Retrieve a plugin from this interface by name.
+
+        Parameters
+        ----------
+        name : str
+          The name the desired plugin.
+
+        Returns
+        -------
+        An object of type ``<interface>Plugin`` where ``<interface>`` is the name of
+        this interface.
+
+        Raises
+        ------
+            Nothing
+        """
+        try:
+            module = find_entry_point(self.entry_point_group, name)
+        except EntryPointError as err:
+            raise PluginError(f"Plugin {name} not found for {self.name} interface")
+        # This assumes that the module's callable is named "callable"
+        if not isinstance(module, Callable):
+            return plugin_module_to_obj(self, module)
+        # If "module" is callable, treat this as a deprecated entry point
+        # whose callable is specified in setup.py directly
+        else:
+            func = module
+            module = import_module(func.__module__)
+            warn(
+                f"Entry point for plugin '{module.__name__}' point to the callable "
+                "rather than the module. This behavior is deprecated. Please update "
+                "the entry point to point to the plugin module. In the future this "
+                "will raise a TypeError.",
+                DeprecationWarning,
+                stacklevel=1,
+            )
+            return plugin_module_to_obj(self, module, module_call_func=func.__name__)
+
+    def get_plugins(self):
+        """Get a list of plugins for this interface."""
+        # This is more complicated than needed. To simplify, we will need to remove the
+        # function name from entry points.
+        plugins = []
+        for ep in get_all_entry_points(self.entry_point_group):
+            module = import_module(ep.__module__)
+            if hasattr(module, "call"):
+                plugins.append(plugin_module_to_obj(module))
+            else:
+                plugins.append(
+                    plugin_module_to_obj(module, module_call_func=ep.__name__)
+                )
+
+        # plugins = [
+        #     self.create_plugin(import_module(ep.__module__),
+        #     module_call_func=ep.__name__)
+        #     for ep in get_all_entry_points(self.entry_point_group)
+        # ]
+        return plugins
+
+    def is_valid(self, name):
+        """Check that an interface is valid.
+
+        Check that the requested interface function has the correct call signature.
+        Return values should be as specified below, but are not programmatically
+        verified.
+
+        Parameters
+        ----------
+        name : str
+          Name of the interface to be validated
+
+        Returns
+        -------
+        bool
+          True if valid, False if invalid
+        '''
+        """
+        plugin = self.get_plugin(name)
```

Comment:
@jsolbrig So does this just raise an error if it is not valid, vs returning "False" ?

And it does not return anything if it is valid?

Reply:
This should return `False` if it is invalid. This isn't well implemented yet. I intend to do this under #72 but figured it was better to get this PR through before making more updates.

Maybe this should be done before the next actual release, though?

Reply:
Probably before the next release - but I'm going to call it good for this PR since it is well documented that it is not fully implemented yet.